src/services/activityReports.js

```diff
@@ -1047,14 +1053,70 @@ export async function createOrUpdate(newActivityReport, report, userId) {
 
   // Approvers are removed if approverUserIds is an empty array
   if (approverUserIds) {
-    await syncApprovers(savedReport.id, approverUserIds);
+    const regionId = savedReport.regionId ?? report?.regionId;
+    let sanitizedApproverIds = [];
+
+    if (regionId) {
+      const permittedApprovers = await usersWithPermissions(
+        [regionId],
+        [SCOPES.APPROVE_REPORTS],
+      );
+      const permittedIds = new Set(permittedApprovers.map((user) => user.id));
+      sanitizedApproverIds = Array.from(
+        new Set(
+          approverUserIds
+            .map((id) => Number(id))
+            .filter((id) => permittedIds.has(id)),
+        ),
+      );
+
+      if (sanitizedApproverIds.length !== approverUserIds.length) {
+        logger.warn('Filtered unauthorized approvers from save request', {
+          namespace,
+          activityReportId: savedReport.id,
+          requestedApproverUserIds: approverUserIds,
+          sanitizedApproverUserIds: sanitizedApproverIds,
+        });
+      }
+    } else {
+      sanitizedApproverIds = approverUserIds;
+      logger.warn('Unable to determine region when syncing approvers; skipping permission filter', {
```

Comment:
Correct me if I'm wrong but if we go to this portion of the else it would mean all the approvers passed in a valid correct? Just not sure I understand the message we log.